exonum/src/blockchain/schema.rs

```diff
@@ -334,6 +356,93 @@ where
     }
 }
 
+/// Location of an isolated call within a block.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize, BinaryValue, ObjectHash,
+)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum CallLocation {
+    /// Call of a transaction within the block.
+    Transaction {
+        /// Zero-based transaction index.
+        index: u64,
+    },
+    /// Call of `before_commit` hook in a service.
+    BeforeCommit {
+        /// Numerical service identifier.
+        id: InstanceId,
+    },
+}
+
+impl ProtobufConvert for CallLocation {
```

Comment:
offtopic: maybe we should add support for enums with named data to `protobuf-convert`.